crates/rpc/src/types/errors.rs

```diff
@@ -0,0 +1,16 @@
+use thiserror::Error;
+use warp::reject::Reject;
+
+#[derive(Error, Debug)]
+pub enum ApiError {
+    #[error("Unauthorized")]
+    Unauthorized,
+
+    #[error("Api Endpoint Not Found: {0}")]
+    NotFound(String),
+
+    #[error("Bad Request: {0}")]
+    BadRequest(String),
+}
+
+impl Reject for ApiError {}
```

Comment:
```suggestion
impl Reject for ApiError {}

```

File is missing a new line at the end 
![image](https://github.com/user-attachments/assets/2cd6645d-6aa3-4def-a481-4590c16e434a)